<?xml version="1.0"?>
<robot name="G35">
  <!-- Base footprint (point on the ground directly below CG) -->
  <link name="base_footprint">
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <geometry>
        <mesh filename="package://vehicle_model_pkg/description/body_2004_Infiniti_G35_Sedan.dae" scale="1 1 1" />
      </geometry>
    </visual>    
  </link>
  
  <!-- Left front wheel -->
  <link name="wheel_left_front">
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <geometry>
        <mesh filename="package://vehicle_model_pkg/description/wheel_2004_Infiniti_G35_Sedan.dae" scale="1 1 1" />
      </geometry>
    </visual>    
  </link>
  <joint name="base_to_lf" type="fixed">
    <parent link="base_footprint"/>
    <child link="wheel_left_front"/>
    <origin xyz="0 0 0"/>
  </joint>

  <!-- Left rear wheel -->
  <link name="wheel_left_rear">
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <geometry>
        <mesh filename="package://vehicle_model_pkg/description/wheel_2004_Infiniti_G35_Sedan.dae" scale="1 1 1" />
      </geometry>
    </visual>    
  </link>
  <joint name="base_to_lr" type="fixed">
    <parent link="base_footprint"/>
    <child link="wheel_left_rear"/>
    <origin xyz="0 0 0"/>
  </joint>

  <!-- Right front wheel -->
  <link name="wheel_right_front">
    <visual>
      <origin xyz="0 0 0" rpy="0 0 3.141"/>
      <geometry>
        <mesh filename="package://vehicle_model_pkg/description/wheel_2004_Infiniti_G35_Sedan.dae" scale="1 1 1" />
      </geometry>
    </visual>    
  </link>
  <joint name="base_to_rf" type="fixed">
    <parent link="base_footprint"/>
    <child link="wheel_right_front"/>
    <origin xyz="0 0 0"/>
  </joint>

  <!-- Right rear wheel -->
  <link name="wheel_right_rear">
    <visual>
      <origin xyz="0 0 0" rpy="0 0 3.141"/>
      <geometry>
        <mesh filename="package://vehicle_model_pkg/description/wheel_2004_Infiniti_G35_Sedan.dae" scale="1 1 1" />
      </geometry>
    </visual>    
  </link>
  <joint name="base_to_rr" type="fixed">
    <parent link="base_footprint"/>
    <child link="wheel_right_rear"/>
    <origin xyz="0 0 0"/>
  </joint>

</robot>